admin/spam/views.py

```diff
@@ -1,96 +1,124 @@
 from __future__ import unicode_literals
 
 from django.shortcuts import render
-from django.contrib.auth.decorators import login_required
 from django.core.mail import send_mail
-from django.http import HttpResponseNotFound
 from django.views.generic import FormView, ListView
-from django.utils.decorators import method_decorator
-from django.core.urlresolvers import reverse
 from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 from website.settings import SUPPORT_EMAIL
 
-from admin.common_auth.logs import update_admin_log, CONFIRM_HAM, CONFIRM_SPAM
+from admin.base.utils import OSFAdmin
+from admin.common_auth.logs import (
+    update_admin_log,
+    CONFIRM_HAM,
+    CONFIRM_SPAM,
+    USER_EMAILED,
+)
 from admin.spam.serializers import serialize_comment
 from admin.spam.forms import EmailForm, ConfirmForm
+from admin.spam.templatetags.spam_extras import reverse_spam_detail
```

Comment:
This would then be defined in base as a template tag and could in fact be used for all current extras that I have defined throughout admin.